libtransact/src/handler/sabre.rs

```diff
@@ -0,0 +1,148 @@
+            .next())
+    }
+
+    /// get_state_entries queries the validator state for data at each of the
+    /// addresses in the given list. The addresses that have been set
+    /// are returned.
+    ///
+    /// # Arguments
+    ///
+    /// * `addresses` - the addresses to fetch
+    fn get_state_entries(
+        &self,
+        addresses: &[String],
+    ) -> Result<Vec<(String, Vec<u8>)>, ContextError> {
+        self.context
+            .get_state_entries(addresses)
+            .map_err(|err| ContextError::ResponseAttributeError(err.to_string()))
+    }
+
+    /// set_state_entry requests that the provided address is set in the validator state to its
+    /// corresponding value.
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - address of where to store the data
+    /// * `data` - payload is the data to store at the address
+    fn set_state_entry(&self, address: String, data: Vec<u8>) -> Result<(), ContextError> {
+        self.context
+            .set_state_entries(vec![(address, data)])
+            .map_err(|err| ContextError::ResponseAttributeError(err.to_string()))
+    }
+
+    /// set_state_entries requests that each address in the provided map be
+    /// set in validator state to its corresponding value.
+    ///
+    /// # Arguments
+    ///
+    /// * `entries` - entries are a hashmap where the key is an address and value is the data
+    fn set_state_entries(&self, entries: Vec<(String, Vec<u8>)>) -> Result<(), ContextError> {
+        self.context
+            .set_state_entries(entries)
+            .map_err(|err| ContextError::ResponseAttributeError(err.to_string()))
+    }
+
+    /// delete_state_entry requests that the provided address be unset
+    /// in validator state. A list of successfully deleted addresses
+    /// is returned.
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - the address to delete
+    fn delete_state_entry(&self, address: &str) -> Result<Option<String>, ContextError> {
+        Ok(self
+            .delete_state_entries(&[address.to_string()])
+            .map_err(|err| ContextError::ResponseAttributeError(err.to_string()))?
+            .into_iter()
+            .next())
+    }
+
+    /// delete_state_entries requests that each of the provided addresses be unset
+    /// in validator state. A list of successfully deleted addresses
+    /// is returned.
+    ///
+    /// # Arguments
+    ///
+    /// * `addresses` - the addresses to delete
+    fn delete_state_entries(&self, addresses: &[String]) -> Result<Vec<String>, ContextError> {
+        self.context
+            .delete_state_entries(addresses)
+            .map_err(|err| ContextError::ResponseAttributeError(err.to_string()))
+    }
+
+    /// add_receipt_data adds a blob to the execution result for this transaction
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - the data to add
+    fn add_receipt_data(&self, _data: Vec<u8>) -> Result<(), ContextError> {
+        unimplemented!()
```

Comment:
Is this intentionally unimplemented? If so, the doc comment should probably address this.

Reply:
ya its not supported in sabre yet, ill update the comment

Reply:
added